sdk/src/Services/DynamoDBv2/Custom/DataModel/_async/Context.Async.cs

```diff
@@ -301,7 +302,7 @@ public Task ExecuteBatchWriteAsync(BatchWrite[] batches, CancellationToken cance
         /// </param>
         /// <param name="operationConfig">Config object which can be used to override that table used.</param>
         /// <returns>AsyncSearch which can be used to retrieve DynamoDB data.</returns>
-        public AsyncSearch<T> ScanAsync<T>(IEnumerable<ScanCondition> conditions, DynamoDBOperationConfig operationConfig = null)
+        public IAsyncSearch<T> ScanAsync<T>(IEnumerable<ScanCondition> conditions, DynamoDBOperationConfig operationConfig = null)
```

Comment:
Changing the return type here is not possible as it would be a breaking change. 
We may have customers with a code base relying on the return type AsyncSearch<T>

Reply:
Sure, have created a new interface so that consumers can choose whether to have an concrete class or an interface returned.

Reply:
You might want to look into using explicit interface implementation https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/interfaces/explicit-interface-implementation

It should allow you to leave the original method untouched but create a second implementation something like

```csharp
IAsyncSearch<T> IAsyncSearch<T>.ScanAsync<T>(IEnumerable<ScanCondition> conditions, DynamoDBOperationConfig operationConfig = null)
{
  ...
}

```
The new method is only visible when working with a variable declared as the interface type.

Reply:
@normj - is someone else picking up this PR?  I was just trying to mock DynamoDBContext with IDynamoDBContext and ran into the same issue (AsyncSearch doesn't implement an interface, and it's constructor is not even public, so it basically makes mocking the context impossible)?  Thanks!  - kirkdavi@